### PR TITLE
Fix shuffleboard not closing

### LINKED ISSUE #666: App does not close correctly
Ever since the Graph update, shuffleboard does not close correctly. This needs to be fixed before we release.

### PR DESCRIPTION
**Pull request by carbotaniuman**
Fixes #666.

### CODE REVIEW

**Review by Daltz333 — APPROVED** [review]

File: app/src/main/java/edu/wpi/first/shuffleboard/app/Shuffleboard.java [hunk 1]
```diff
@@ -139,16 +139,21 @@ public void start(Stage primaryStage) {
...
           } catch (IOException ex) {
             logger.log(Level.WARNING, "Could not save the layout", ex);
           }
-        } else if (bt == ButtonType.CANCEL) {
-          // cancel the close request by consuming the event
-          closeEvent.consume();
         }
+
+        Platform.exit();
         // Don't need to check for NO because it just lets the window close normally
```
Comment (sciencewhiz) [inline app/src/main/java/edu/wpi/first/shuffleboard/app/Shuffleboard.java:157]: should the comment go above exit command?

Comment (carbotaniuman) [reply app/src/main/java/edu/wpi/first/shuffleboard/app/Shuffleboard.java]: Fixed.

**Comment by AustinShalit** [comment]
So Platform should auto close when all stages & threads close. This is more of a patch than the real fix. Do you have any ideas what thread the graphing API still has running that is not being closed?

**Comment by carbotaniuman** [comment]
I checked out the thread list after this, and every thread left was either a JDK or a daemon thread, which shouldn't impede shutdowns.

**Comment by carbotaniuman** [comment]
With chart-fx:
image

Without:
image

It's a real ???? right now.

**Comment by carbotaniuman** [comment]
@AustinShalit debugged it - https://github.com/GSI-CS-CO/chart-fx/blob/8485ed6cc44f2eb531cfe55c10411c0a1da35c6e/chartfx-chart/src/main/java/de/gsi/chart/utils/FXUtils.java#L39

**Review by AustinShalit — APPROVED** [review]

**Pull request merged**